File: Primes-In-Numbers/PrimesInNumbers.py
def get_factors(n):
    factors = {}
    divisor = 2
    while n >= 2:
        if n % divisor == 0:
          if divisor not in factors:
              factors[divisor] = 0
          factors[divisor] = factors[divisor] + 1
          n = n / divisor
        else:
            divisor += 1
    return factors
  
def prime_factors(n):
    answer = ''
    obj = get_factors(n)
    for key, value in obj.items():
      if value < 2:
        answer += f'({key})'
      else:
        answer += f'({key}**{value})'
    return answer
  

File: Primes-In-Numbers/PrimesInNumbers.js
function getFactors(n) {
  const factors = {};
  let divisor = 2;
  while (n >= 2) {
    if (n % divisor === 0) {
      factors[divisor] = ++factors[divisor] || 1;
      n = n / divisor;
    } else {
      divisor++;
    }
  }
  return factors;
}

function primeFactors(n) {
  let answer = "";
  const obj = getFactors(n);
  for (const key in obj) {
    answer += obj[key] < 2 ? `(${key})` : `(${key}**${obj[key]})`;
  }
  return answer;
}

const randomNumber = Math.floor(Math.random() * 10000);
console.log(randomNumber, "=>", primeFactors(randomNumber));
